crates/solver/src/settlement_submission/submitter.rs

```diff
@@ -13,91 +13,109 @@
 // from outside) so it is only at that point that we need to check the hashes individually to the
 // find the one that got mined (if any).
 
-use super::{flashbots_api::FlashbotsApi, ESTIMATE_GAS_LIMIT_FACTOR};
-use crate::settlement::Settlement;
+use super::ESTIMATE_GAS_LIMIT_FACTOR;
+use crate::{interactions::block_coinbase, settlement::Settlement};
 use anyhow::{anyhow, ensure, Context, Error, Result};
 use contracts::GPv2Settlement;
-use ethcontract::{transaction::Transaction, Account};
+use ethcontract::{contract::MethodBuilder, dyns::DynTransport, transaction::Transaction, Account};
 use futures::FutureExt;
 use gas_estimation::{EstimatedGasPrice, GasPriceEstimating};
 use primitive_types::{H256, U256};
 use shared::Web3;
 use std::time::{Duration, Instant, SystemTime};
 use web3::types::TransactionReceipt;
 
-pub struct FlashbotsSolutionSubmitter<'a> {
+/// Parameters for transaction submitting
+#[derive(Clone, Default)]
+pub struct SubmitterParams {
+    /// Desired duration to include the transaction in a block
+    pub target_confirm_time: Duration, //todo change to blocks in the following PR
+    /// Estimated gas consumption of a transaction
+    pub gas_estimate: U256,
+    /// Maximum max_fee_per_gas to pay for a transaction
+    pub gas_price_cap: f64,
+    /// Maximum duration of a single run loop
+    pub deadline: Option<SystemTime>,
+    /// There are two ways to pay for a transaction:
+    /// 1. Set valid gas price (default)
+    /// 2. Set 0 gas price, but add additional step to settlement to pay directly to the coinbase
+    /// Contains quantity of gas needed to pay for the additional step
+    pub pay_gas_to_coinbase: Option<U256>,
```

Comment:
In this case I would like this configuration to be an enum to make it clear which parameters are valid together.

Reply:
Its not convenient to couple this with gasPrice because:
1. Gas price is unknown at the time when this is defined
2. Gas price is mutable value because of the iterating run loop.
Tried but ends with even less clean code...

Reply:
What I was confused about is the comment "Set 0 gas price" . I thought this was referring to something in the struct so I suggested an enum that either stores that or pay_gas_to_coinbase. I see now that this isn't the case. I think the comment is talking about what we (this code) do with the transaction so it makes sense.

Reply:
Good point. We have to double check how Archer works today. Could even say we should delete the implementation for now and merge Flashbots + Traditional workflow first then maybe re-add archer later?

Reply:
I think `priority_fee` is largely equivalent to paying to coinbase right? At least if the `max_fee_per_gas` is large enough. I would be in favour of removing it if that is the case.

Reply:
With the new archer (eden) integration this field is not needed anymore.